indexing-service/src/main/java/org/apache/druid/indexing/overlord/ZkWorker.java

```diff
@@ -73,19 +74,18 @@ public ZkWorker(Worker worker, PathChildrenCache statusCache, final ObjectMapper
   {
     return (ChildData input) -> {
       try {
-        JsonNode parent = jsonMapper.readTree(input.getData());
-        if (parent != null) {
-          JsonNode id = parent.get(TaskAnnouncement.TASK_ID_KEY);
-          if (id != null) {
-            return id.asText();
+        JsonParser parser = jsonMapper.getFactory().createParser(input.getData());
```

Comment:
Use try-with-resources statement to ensure parser is closed.

try (JsonParser parser = jsonMapper.getFactory().createParser(input.getData())) {

Reply:
Fixed in 2360cd3963